test/parallel/test-dotenv-edge-cases.js

```diff
@@ -35,4 +35,16 @@ describe('.env supports edge cases', () => {
     assert.strictEqual(child.code, 0);
   });
 
+  it('should not override existing environment variables', async () => {
+    const code = `
+      require('assert').strictEqual(process.env.BASIC, 'existing');
+    `.trim();
+    const child = await common.spawnPromisified(
+      process.execPath,
+      [ `--env-file=${validEnvFilePath}`, '--eval', code ],
+      { cwd: __dirname, env: { BASIC: 'existing' } },
+    );
+    assert.strictEqual(child.stderr, '');
+    assert.strictEqual(child.code, 0);
+  });
```

Comment:
Can we add a test that variables can be mixed, with environment-defined ones taking precedence but not-present-in-environment ones still getting set? Like say `.env` contains:

```shell
A=1
B=2
```

When run as `A=3 node --env-file=.env --print 'process.env.A + ',' + process.env.B'` I would expect to see `3,2`.

Reply:
Added exactly that test. Thought it was better to add another example .env file than to play with the intentionally mangled vars in `fixtures/valid.env`.